### ISSUE TITLE
Cairo error when trying to export an annotated pdf

### ISSUE THREAD

**Issue by mmaluff** [xournalpp/xournalpp#5948]
### Operating System

Linux

### (Linux only) Distribution

Pop 22.04

### (Linux only) Desktop Environment

GNOME

### (Linux Only) Display Server

X.Org

### Installation Method

Flatpak

### Xournal++ Version

1.2.3

### libgtk Version

3.24.43

### Bug Description

When trying to export annotated PDFs to PDF, I get the following error:

Error while finalizing the PDF Cairo surface
Cairo error: out of memory

![image](https://github.com/user-attachments/assets/cdf9703d-0128-4363-9b5e-7286dad5852d)

It then outputs a corrupted pdf that can't be opened.

### Expected Behaviour

A pdf is correctly exported.

### Steps to Reproduce

1. Open any pdf document
2. Make a note or highlight
3. Export it to pdf with a new name

### Additional Context

The error does not seem to happen when I take notes on a blank page and export to pdf, only when editing an existing pdf.

**Comment by jschlatow** [comment]
I ran into the same problem but with a single-page 100K document using version 1.2.3. I was able to export the pdf on another system with version 1.2.2 installed, though.

**Comment by jschlatow** [comment]
> Does it give you the same error if you don't annotate at all (so just open the pdf and export it)?

Yes, same error for me.

**Comment by mmaluff** [comment]
My pdfs are relatively small, the smallest I've seen this this issue with is 307kB, though I haven't been able to reproduce it with much smaller pdfs. They have no pictures, just text; I mainly read philosophy papers. I am also able to reproduce this with no annotation, just opening the pdf and exporting it.

**Comment by jschlatow** [comment]
I tried this today on another system (Archlinux) that also has xournal 1.2.3 installed. Surprised that the export worked on this system, I investigated. The broken system runs Manjaro, which uses the same binaries, yet, has a slightly different cairo package installed. I found the following bug report related to this cairo version: https://gitlab.archlinux.org/archlinux/packaging/packages/cairo/-/issues/2

**Comment by bjacques** [comment]
> @jschlatow Are you also using the flatpak version? It may be an issue with the Cairo version bundled in there.

Good call, uninstalling Flatpak version and installing DEB package resolved this for me. (PopOS)

**Comment by Nicepawer7** [comment]
Uninstalling flatpak version and installing xuornalpp worked for me too (arch)

**Comment by ngasull** [comment]
Got exactly the same with nix in 1.2.4 and 1.2.5, which builds xournal++ from source. Nix is a package manager that makes consistent builds. 

@bhennion can you confirm it is related to #5847 ? It could help fix the Flatpak and the nix package :smiley: 

Related Nixpkgs issue: https://github.com/NixOS/nixpkgs/issues/365575

**Comment by rubyFeedback** [comment]
Thanks, that information helped me too.

**Comment by cnschu** [comment]
there is no new release of cairo available, and no way to report this bug or request a new release with this bug fixed.
any ideas?

**Comment by j0nath4nl1k3-crypto** [comment]
Hello, I'm translating this text into English with the help of ChatGPT since I'm not fluent in English.

I ran into the same error and managed to work around it in a simple way.
In Xournal++, I did the following:
Pressed Ctrl + P (or just went to Print), and printed one page at a time.
For example, my file had 10 pages, so I repeated this process 10 times — one for each page.

After that, I had 10 separate PDF files. Then, I used a program called PDFsam (but you can use any tool that can merge PDFs) to combine them all into a single document.

It’s kind of a workaround, but it worked for me — I hope it helps someone else too.

**Comment by Krabbenhoft** [comment]
I have this issue on Windows 11 v25H2 with Xournal++ with a very small document.

**Comment by hrishikeshrt** [comment]
I got same issue with Cairo 1.18.4 on MacOS, Xournal++ 1.2.8
